tests/test-sessions/test-file-sessions/src/test/java/org/eclipse/jetty/server/session/FileTestHelper.java

```diff
@@ -223,28 +218,18 @@ public static boolean checkSessionPersisted (SessionData data)
             assertEquals(data.getExpiry(), expiry);
             assertEquals(data.getMaxInactiveMs(), maxIdle);
 
-            Map<String,Object> attributes = new HashMap<>();
-            
-            int size = di.readInt();
-            if (size > 0)
-            {
-               ClassLoadingObjectInputStream ois =  new ClassLoadingObjectInputStream(di);
-                for (int i=0; i<size;i++)
-                {
-                    String key = ois.readUTF();
-                    Object value = ois.readObject();
-                    attributes.put(key,value);
-                }
-            }
+            SessionData tmp = new SessionData(id, contextPath, vhost, created, accessed, lastAccessed, maxIdle);
+            ClassLoadingObjectInputStream ois =  new ClassLoadingObjectInputStream(di);
```

Comment:
Closing the CLOIS and  it is wrapping another stream, which should not be closed at that point.

Reply:
Same as above, you can initialize all these streams in the initial try-with-resources?
```java
try (FileInputStream in = new FileInputStream(file);
        DataInputStream di = new DataInputStream(in);
        ClassLoadingObjectInputStream ois =  new ClassLoadingObjectInputStream(di)) 
{
    ...
}
```

Reply:
No, that cannot be done. When the CLOIS is created, it tries to read some bytes from the stream. It's not correct for those bytes to be read at that point, as it will screw up the deserialization.